### DISCUSSION TITLE
How to populate select box from JSON API response

### DISCUSSION THREAD

**Discussion by kharabela** [bigskysoftware/htmx#2815] [Q&A]
I have an API that returns axios JSON response
  axios.get('/countries')
   .then(function (response) {
        const obj = JSON.parse(response.data);
      //  response.data.forEach(console.log(count.name));
 console.log(response.data);
}

This returns sample JSON response as below
[
    {
        "name": "Afghanistan",
        "code": "AF"
    },
    {
        "name": "Åland Islands",
        "code": "AX"
    },
]

For each item of response data how to populate this in html select drop down

**Comment by Telroshan** [comment]
Hey, you'll want to do standard DOM manipulations and for example, generate `options` tag with name/value from the data you retrieve, then `appendChild` them to your select element.

Please note that this is not related to htmx at all, as you're fetching data with `axios` and not htmx, retrieving JSON instead of HTML to swap, so htmx will likely not be of any use here.
